x-pack/plugins/entity_manager/public/lib/entity_client.test.ts

```diff
@@ -0,0 +1,175 @@
+
+const commonEntityFields: EntityLatest = {
+  entity: {
+    last_seen_timestamp: '2023-10-09T00:00:00Z',
+    id: '1',
+    display_name: 'entity_name',
+    definition_id: 'entity_definition_id',
+  } as EntityLatest['entity'],
+};
+
+describe('EntityClient', () => {
+  let entityClient: EntityClient;
+
+  beforeEach(() => {
+    entityClient = new EntityClient(coreMock.createStart());
+  });
+
+  describe('asKqlFilter', () => {
+    it('should return the value when indentity_fields is a single string', () => {
+      const entityLatest: EntityLatest = {
+        entity: {
+          ...commonEntityFields.entity,
+          identity_fields: ['service.name', 'service.environment'],
+          type: 'service',
+        },
+        service: {
+          name: 'my-service',
+        },
+      };
+
+      const result = entityClient.asKqlFilter(entityLatest);
+      expect(result).toEqual('service.name: my-service');
+    });
+
+    it('should return values when indentity_fields is composed by multiple fields', () => {
+      const entityLatest: EntityLatest = {
+        entity: {
+          ...commonEntityFields.entity,
+          identity_fields: ['service.name', 'service.environment'],
+          type: 'service',
+        },
+        service: {
+          name: 'my-service',
+          environment: 'staging',
+        },
+      };
+
+      const result = entityClient.asKqlFilter(entityLatest);
+      expect(result).toEqual('(service.name: my-service AND service.environment: staging)');
+    });
+
+    it('should return identity fields values when an indentity field value is an array', () => {
+      const entityLatest: EntityLatest = {
+        entity: {
+          ...commonEntityFields.entity,
+          identity_fields: ['service.name', 'service.environment'],
+          type: 'service',
+        },
+        service: {
+          name: 'my-service',
+          environment: ['prod', 'staging', 'dev'],
+        },
+      };
+
+      const result = entityClient.asKqlFilter(entityLatest);
+      expect(result).toEqual(
+        '(service.name: my-service AND (service.environment: prod OR service.environment: staging OR service.environment: dev))'
+      );
+    });
+
+    it('should throw an error when identity fields are missing', () => {
+      const entityLatest: EntityLatest = {
+        ...commonEntityFields,
+      };
+
+      expect(() => entityClient.asKqlFilter(entityLatest)).toThrow('Identity fields are missing');
+    });
+
+    it('should ignore fields that are not present in the entity', () => {
+      const entityLatest: EntityLatest = {
+        entity: {
+          ...commonEntityFields.entity,
+          identity_fields: ['host.name', 'foo.bar'],
+        },
+        host: {
+          name: 'my-host',
+        },
+      };
+
+      const result = entityClient.asKqlFilter(entityLatest);
+      expect(result).toEqual('host.name: my-host');
+    });
+  });
+
+  describe('getIdentityFieldsValue', () => {
```

Comment:
You don't really need this test, you're already testing all of these paths through the upper layer, so you can hide this code from testing while still testing the same requirements and leaving your implementation more flexible.

Reply:
This function can be called independently. I assume it's safer to ensure it works with these tests.

Reply:
Ah, is it also exported from the client independently? If so, then yes, let's keep the tests but perhaps those two functions should live in different files too?

Reply:
yeah. sounds good. Move them to another file but export them in `EntityClient`, right?

Reply:
Expose it on the client if you believe it will be useful, once it's added it's harder to remove.